src/tests/roc_audio/test_depacketizer.cpp

```diff
@@ -594,5 +627,34 @@ TEST(depacketizer, timestamp) {
     }
 }
 
+TEST(depacketizer, read_after_error) {
+    const status::StatusCode codes[] = {
+        status::StatusUnknown,
+        status::StatusNoData,
+    };
+
+    for (unsigned n = 0; n < ROC_ARRAY_SIZE(codes); ++n) {
+        PcmEncoder encoder(PcmFmt, SampleSpecs);
+        PcmDecoder decoder(PcmFmt, SampleSpecs);
+
+        packet::Queue queue;
+        StatusReader reader(queue, codes[n]);
+        Depacketizer dp(reader, decoder, SampleSpecs, false);
+        CHECK(dp.is_valid());
+
+        queue.write(new_packet(encoder, 0, 0.11f, Now));
+
+        reader.enable();
+        expect_output(dp, SamplesPerPacket, 0.00f, 0);
+
+        CHECK(dp.is_valid());
+
+        reader.disable();
+        expect_output(dp, SamplesPerPacket, 0.11f, Now);
```

Comment:
After this piece:

```cpp
        reader.enable();
        expect_output(dp, SamplesPerPacket, 0.00f, 0);
```

and after this:

```cpp
        reader.disable();
        expect_output(dp, SamplesPerPacket, 0.11f, Now);
```

I suggest to check that reader.read() was actually called. E.g. we can add reader.call_count() and check it.